V1Michigan/v1
# Finish TypeScript migration

### Status
**READY**

### Description
* Fix a few changes that were lost between #2 and #3
* Remove old unused `components/*.js` files
* Convert rest of `.js` files to `.tsx`
* Add linting for Next + React hooks

Still not actually checking types with `strict` mode, [made another branch here](https://github.com/V1Michigan/v1/tree/ts-strict) so we can start working on those errors (not sure it's a priority though).

### Related PRs
Follow-up to #3 


### Dependencies
- ADDED `@typescript-eslint/eslint-plugin`
- ADDED `eslint-config-next`

### Migrations
NO

### Steps to Test or Reproduce

```sh
git pull --prune
git checkout ts-migration-part-2
yarn install
npm run dev
npm run lint:check
```
